src/main/java/org/HdrHistogram/AbstractHistogram.java

```diff
@@ -308,13 +312,22 @@ private void init(final long lowestDiscernibleValue,
         subBucketHalfCount = subBucketCount / 2;
         subBucketMask = ((long)subBucketCount - 1) << unitMagnitude;
 
+        if (subBucketCountMagnitude + unitMagnitude > 62) {
+            // subBucketCount entries can't be represented, with unitMagnitude applied, in a positive long.
+            // Technically it still sort of works if their sum is 63: you can represent all but the last number
+            // in the shifted subBucketCount. However, the utility of such a histogram vs ones whose magnitude here
+            // fits in 62 bits is debatable, and it makes it harder to work through the logic.
+            // Sums larger than 64 are totally broken as leadingZeroCountBase would go negative.
+            throw new IllegalArgumentException("Cannot represent numberOfSignificantValueDigits worth of values " +
+                    "beyond lowestDiscernibleValue");
+        }
 
         // determine exponent range needed to support the trackable value with no overflow:
         establishSize(highestTrackableValue);
 
         // Establish leadingZeroCountBase, used in getBucketIndex() fast path:
         // subtract the bits that would be used by the largest value in bucket 0.
-        leadingZeroCountBase = 64 - unitMagnitude - subBucketHalfCountMagnitude - 1;
+        leadingZeroCountBase = 64 - unitMagnitude - subBucketCountMagnitude;
```

Comment:
Now that `subBucketCountMagnitude` is always `subBucketHalfCountMagnitude + 1`, might as well use it here.